### PR TITLE
gdImageGdPtr memory leak

### PR DESCRIPTION
**Pull request by me22bee**
discussed in issue #700.

### CODE REVIEW

**Comment by pierrejoye** [comment]
Thank you :)

File: src/gd_gd.c [hunk 1]
```diff
@@ -400,8 +400,11 @@ BGD_DECLARE(void *) gdImageGdPtr (gdImagePtr im, int *size)
 	void *rv;
 	gdIOCtx *out = gdNewDynamicCtx (2048, NULL);
 	if (out == NULL) return NULL;
-	_gdImageGd (im, out);
-	rv = gdDPExtractData (out, size);
+	if (_gdImageGd (im, out)) {
```
Comment (cmb69) [inline src/gd_gd.c:403]: @pierrejoye, `_gdImageGd()` is a void function; checking its return value doesn't make sense, and likely causes a compile failure or warning. I think we need to setup CI (building and running the tests) to catch such issues.

Comment (pierrejoye) [reply src/gd_gd.c:403]: Thanks @cmb69!

Damned, misread the signature, sorry :/

Comment (me22bee) [reply src/gd_gd.c:403]: sorry, my mistake :(
@pierrejoye can you please revert this?

Comment (pierrejoye) [reply src/gd_gd.c:403]: Reverted. Not that critical as the GD format is really really for dev/testing only. However open for an improved fix :)

Also, as @cmb69 mentioned, GD/GD2 formats are going to be deprecated and removed for 3.0

**Pull request merged**